spec/models/grant_submission_response_spec.rb

```diff
@@ -9,22 +10,72 @@
   it { is_expected.to respond_to(:decimal_val) }
   it { is_expected.to respond_to(:datetime_val) }
   it { is_expected.to respond_to(:boolean_val) }
+  it { is_expected.to respond_to(:document) }
 
   describe '#validations' do
     let(:grant)       { create(:published_open_grant_with_users)}
     let(:other_grant) { create(:published_open_grant_with_users)}
     let(:submission)  { create(:grant_submission_submission, grant: grant, form: grant.form)}
-    let(:response)    { create(:string_val_response, submission: submission,
-                                                     question: grant.form.questions.where(response_type: 'short_text').first)}
 
-    it 'validates response to a question from the grant' do
-      expect(response).to be_valid
+    context 'string_val' do
+      let(:response)    { create(:string_val_response, submission: submission,
+                                                       question:   grant.form.questions.where(response_type: 'short_text').first)}
+
+      context '#validations' do
+        it 'validates response to a question from the grant' do
+          expect(response).to be_valid
+        end
+
+        it 'validates if response is to a question from another grant' do
+          response.update_attribute(:grant_submission_question_id, other_grant.form.questions.where(response_type: 'short_text').first.id)
+          expect(response).not_to be_valid
+          expect(response.errors).to include(:grant_submission_question_id)
+        end
+
+        it 'validates length' do
+          response.update_attribute(:string_val, Faker::Lorem.characters(number: 256))
+          expect(response).not_to be_valid
+          expect(response.errors).to include(:string_val)
+        end
+
+        it 'validates if attachment' do
```

Comment:
Maybe this is overkill but maybe have this covered for all invalid response_type.  Seems to be only applied for 'short_text'

Reply:
Yep. Will open this as a new ticket and add specs for other response types.

Reply:
See #291